### DISCUSSION TITLE
Fundamentos da Orientação a Objetos - Aula Events

### DISCUSSION THREAD

**Discussion by Rafa-Matheus** [balta-io/forum#431] [Tira Dúvidas] [answered]
Como vai, Balta?
Gostaria de pedir, caso seja possível, para revisar ou repassar o conteúdo da aula Events no módulo de Orietação a Objetos.
Fiquei confuso pois você criou dois métodos manipuladores OnRoomSoldOut, depois dentro de um deles atribuiu um manipulador a outro manipulador para invocar o método, tbm não ficou claro onde o delegate entra naquilo td, enfim... não entendi direito... pode me explicar ou revisar o conteúdo desta aula?

**Comment by andrebaltieri** [comment, marked as answer]
Boa tarde, @Rafa-Matheus, como vai?

Vou preparar um conteúdo sobre o assunto, valeu pelo feedback!

**Reply by Rafa-Matheus** [reply]
Obrigado, Balta. Se possível, poderia também incluir Actions, Funcs e Predicates na aula, associados a eventos? E ainda nesse contexto ensinar sobre métodos anônimos? Se for possível, seria bem legal! 
Vlw!

**Comment by Lucas-back-end** [comment]
Olá, Grande Balta. 

Então, refiz a aula duas vezes com anotações mas mesmo assim fiquei confuso na **aula de  Events.** Como eu consigo o código da aula para eu tentar fazer uma observação melhor e tentar ligar os pontos? Eu consegui entender até determinado tempo. Depois fiquei perdido. Mas de qualquer forma, irei procurar por outtros vídeos que falam sobre o assunto.

**Comment by andrebaltieri** [comment]
<img width="435" alt="Screenshot 2024-03-06 at 12 55 47" src="https://github.com/balta-io/forum/assets/965305/3733c029-ba0a-4103-ac6e-6d9d4883d88f">
Boa tarde, @Lucas-back-end, tudo bem?
Embaixo de cada aula tem o link para o código fonte!

**Reply by Lucas-back-end** [reply]
Eu até olhei o link mas não encontrei o código. Vou conferir melhor agora. Grato!

**Comment by ThiagoBoccalon** [comment]
Olá Balta, espero que você esteja bem.

Semelhante aos colegas de mensagens a cima eu também fiquei um pouco confuso com relação a esse conteúdo.
Você chegou a criar algum outro material explicando o mesmo assunto:


Vlw.

**Reply by andrebaltieri** [reply]
Show!! Pessoal, estou cobrindo melhor esta parte de eventos no próximo curso que sai em Novembro!!! 🚀

**Comment by victorrj99** [comment]
> Como vai, Balta?
> 
> Gostaria de pedir, caso seja possível, para revisar ou repassar o conteúdo da aula Events no módulo de Orietação a Objetos.
> 
> Fiquei confuso pois você criou dois métodos manipuladores OnRoomSoldOut, depois dentro de um deles atribuiu um manipulador a outro manipulador para invocar o método, tbm não ficou claro onde o delegate entra naquilo td, enfim... não entendi direito... pode me explicar ou revisar o conteúdo desta aula?

Boa tarde balta, tenho o mesmo problema do amigo, por muitas vezes sua webcam tapa o codigo e não foi explicado sobre o object sender, mas seria sobre eventos em específico
